### PR TITLE
add changes to integrate dokka validation

### PR DESCRIPTION
**Pull request by abhishek1508**
## Description

Add changes to integrate dokka validation for UI SDK

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

Add changes to integrate dokka validation for UI SDK

### Implementation

Add changes to integrate dokka validation for UI SDK

## Screenshots or Gifs

Please include all the media files to give some context about what's being implemented or fixed. It's not mandatory to upload screenshots or gifs, but for most of the cases it becomes really handy to get into the scope of the feature / bug being fixed and also it's REALLY useful for UI related PRs screenshot gif

## Testing

Please describe the manual tests that you ran to verify your changes

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/legacy/SdkVersionChecker.kt [hunk 1]
```diff
@@ -2,9 +2,12 @@ package com.mapbox.navigation.ui.legacy
 
 class SdkVersionChecker(private val currentSdkVersion: Int) {
 
-    fun isGreaterThan(sdkCode: Int): Boolean =
-        currentSdkVersion > sdkCode
-
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/legacy/SdkVersionChecker.kt:7]: For documentation purposes noting that this breaks SemVer cc @langsmith

Comment (abhishek1508) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/legacy/SdkVersionChecker.kt:7]: Added it to the Migration Docs for UI SDK

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/legacy/NavigationConstants.kt [hunk 2]
```diff
@@ -158,60 +158,214 @@ object NavigationConstants {
...
     const val MAP_DATABASE_PATH_KEY = "offline_map_database_path_key"
+
+    /**
+     * Bundle key to store map style URL
+     */
     const val MAP_STYLE_URL_KEY = "offline_map_style_url_key"
 
-    // Step Maneuver Types
+    /**
+     * Indicates "tuen" maneuver type to be rendered in the [com.mapbox.navigation.ui.instruction.maneuver.ManeuverView]
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/legacy/NavigationConstants.kt:192]:
```suggestion
     * Indicates "turn" maneuver type to be rendered in the [com.mapbox.navigation.ui.instruction.maneuver.ManeuverView]
```

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/legacy/NavigationConstants.kt [hunk 3]
```diff
@@ -244,6 +401,9 @@ object NavigationConstants {
     )
     annotation class ManeuverModifier
 
+    /**
+     * Rounding Increments for use in [com.mapbox.navigation.core.MapboxDistanceFormatter].
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/legacy/NavigationConstants.kt:405]:
```suggestion
     * Rounding increments for use in [com.mapbox.navigation.core.MapboxDistanceFormatter].
```

**Pull request merged**